taichi/transforms/alg_simp.cpp

```diff
@@ -28,6 +28,14 @@ class AlgSimp : public BasicStmtVisitor {
       : BasicStmtVisitor(), fast_math(fast_math_) {
   }
 
+  void visit(UnaryOpStmt *stmt) override {
+    if (stmt->is_cast() &&
+        stmt->cast_type == stmt->operand->ret_type.data_type) {
```

Comment:
Maybe we should test `stmt->op` as well? There are `cast_value` and `cast_bits` :-)

Reply:
If `stmt->cast_type == stmt->operand->ret_type.data_type`, I think the result should always be the same as the input no matter it's `cast_value` or `cast_bits`.